### ISSUE TITLE
Installing ANCOMBC

### ISSUE THREAD

**Issue by AritriSanyal** [FrederickHuangLin/ANCOMBC#251]
Hi! I have been trying to install the AMCOMBC package on R (v. 4.3.2) using Bioconductor (3.18; BiocManager v.1.30.22). 
I have been unable to get the package installed using 
if (!require("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install(version = "3.18")
BiocManager::install("ANCOMBC") (withoutforce = TRUE) 

The issue is:
'getOption("repos")' replaces Bioconductor standard repositories, see 'help("repositories", package = "BiocManager")' for
details.
Replacement repositories:
    CRAN: https://cran.rstudio.com/
Bioconductor version 3.18 (BiocManager 1.30.22), R 4.3.2 (2023-10-31 ucrt)
Installation paths not writeable, unable to update packages
  path: C:/Program Files/R/R-4.3.2/library
  packages:
    cluster, foreign, lattice, MASS, Matrix, mgcv, nlme, rpart
Error: attempt to apply non-function
In addition: Warning message:
package(s) not installed when version(s) same as or greater than current; use `force = TRUE` to re-install: 'ANCOMBC'

I have separately installed the packages mentioned in the error (cluster, foreign, lattice, MASS, Matrix, mgcv, nlme, rpart) and checked for their updates, but this issue still persists. Also tried installing using BiocManager::install("ANCOMBC") as well.

I have even tried installing the package on R (v. 4.3.0), however the BiocManager version is creating an issue and not working even though I installed Bioconductor version 3.12 version. 

Would be grateful for any help with what I may be doing wrong! Thanks!

**Comment by JiatingH** [comment]
So would you mind to share that how to deal with these bugs? I am also struggling with the installation of ANCOMBC in the MAC pro

**Comment by AritriSanyal** [comment]
I am not entirely sure what finally worked, but I removed all the packages that was been shown as errors for updating. Also for some of the packages I removed the files from the R library folder on my computer to remove it completely. Then I installed the ANCOMBC package, and it was downloaded, then after that while loading the package, I was getting errors for installing the dependent packages, which I added and now I am able to use ANCOMBC.
It maybe that the packages updating/installing through the downloader was an older version, which was conflicting with the installation process, however I am not very certain of this.
